src/workerd/io/docker-client.c++

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "docker-client.h"
+#include <kj/debug.h>
+#include <kj/encoding.h>
+#include <workerd/util/http-util.h>
+
+namespace workerd::io {
+
+DockerClient::DockerClient(kj::HttpClient& httpClient)
+    : httpClient(httpClient), dockerSocketPath("/var/run/docker.sock") {}
+
+kj::Promise<kj::String> DockerClient::makeDockerRequest(
+    kj::HttpMethod method, kj::StringPtr path, kj::Maybe<kj::String> body) {
+  
+  auto url = kj::str("http://localhost", path);
+  auto headerTable = kj::heap<kj::HttpHeaderTable>();
+  auto headers = kj::HttpHeaders(*headerTable);
+  auto request = httpClient.request(method, url, headers);
+  
+  KJ_IF_SOME(b, body) {
+    auto stream = kj::heap<kj::StringInputStream>(kj::mv(b));
+    request.body = stream.attach(kj::mv(stream));
+  }
+  
+  auto response = co_await request.response;
+  auto responseBody = co_await response.body->readAllText();
+  
+  if (response.statusCode >= 400) {
+    KJ_FAIL_REQUIRE("Docker API error", response.statusCode, responseBody);
```

Comment:
Does that throw?
We probably need a more robust error handling.
I'm not sure if we should look for an existing C++ Docker client or if it makes sense to build one on top of KJ?

Reply:
I think the response to my first question is in:

```c++
  try {
    co_await makeDockerRequest(kj::HttpMethod::GET, kj::str("/containers/", containerId, "/json"));
    co_return true;
  } catch (kj::Exception& e) {
    co_return false;
  }
```